src/server/scripts/Spells/spell_hunter.cpp

```diff
@@ -270,6 +326,282 @@ class spell_hun_disengage : public SpellScriptLoader
+                        GetCaster()->GetSpellHistory()->ModifyCooldown(spellCds.at(selector), value);
+                    else
+                        GetCaster()->GetSpellHistory()->ModifyCharge(sSpellMgr->GetSpellInfo(spellCds.at(selector))->ChargeCategoryId, value);
+                }
+
+                if (GetCaster()->HasAura(SPELL_HUNTER_ASPECT_BEAST))
+                {
+                    Unit *target = GetHitUnit();
+                    Pet *pet = GetCaster()->ToPlayer()->GetPet();
+                    uint16 spec = GetCaster()->GetGuardianPet()->ToPet()->GetSpecialization();
+
+                    switch (spec)
+                    {
+                    case 81:    // tenacity = 81
+                        pet->CastSpell(target, SPELL_HUNTER_BESTIAL_TENACITY, true);
+                        break;
+                    case 79:   // cunning = 79
+                        pet->CastSpell(target, SPELL_HUNTER_BESTIAL_CUNNING, true);
+                        break;
+                    case 74:  // ferocity = 74
+                        pet->CastSpell(target, SPELL_HUNTER_BESTIAL_FEROCITY, true);
+                        break;
+                    }
+                }
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_hun_flanking_strike_SpellScript::CalculateDamage, EFFECT_0, SPELL_EFFECT_WEAPON_PERCENT_DAMAGE);
+                OnCast += SpellCastFn(spell_hun_flanking_strike_SpellScript::HandleOnCast);
+                OnCheckCast += SpellCheckCastFn(spell_hun_flanking_strike_SpellScript::CheckCast);
+                AfterHit += SpellHitFn(spell_hun_flanking_strike_SpellScript::HandleAfterHit);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_hun_flanking_strike_SpellScript();
+        }
+};
+
+// 204740 - Flanking Strike pet
+class spell_hun_flanking_strike_pet : public SpellScriptLoader
+{
+    public:
+        spell_hun_flanking_strike_pet() : SpellScriptLoader("spell_hun_flanking_strike_pet") { }
+
+        class spell_hun_flanking_strike_pet_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_hun_flanking_strike_pet_SpellScript);
+
+            void CalculateDamage(SpellEffIndex /*effIndex*/)
+            {
+                if (Unit *target = GetHitUnit())
+                {
+                    if(Unit *caster = GetCaster())
+                        if (Unit *owner = GetCaster()->GetOwner())
+                        {
+                            int32 ap = owner->ToPlayer()->GetInt32Value(UNIT_FIELD_ATTACK_POWER);
+                            float vers = owner->ToPlayer()->GetFloatValue(PLAYER_VERSATILITY);
+                            int32 level = caster->getLevel();
+                            //formula taken from tooltip
+                            float result = 1.5*(ap*2.152)*(1 + vers)*(std::min(20, level)*0.05f);
```

Comment:
Do you mean that versatility shouldn't be part of the formula because it's applied on every spell that deals damage? If that's true it's not longer needed on the formula.